test/test_location.js

```diff
@@ -188,47 +189,128 @@ describe('states.location', function() {
+                        state:'states:end'
+                    })
+                    .check(function(api) {
+                        assert(
+                            address_chosen instanceof AddressResult,
+                            "Expected an AddressResult");
+                        assert.strictEqual(
+                            address_chosen.label,
                             'Friend Street, Boston, MA 02114, USA');
-                })
-                .run();
+                    })
+                    .run();
+            });
         });
 
-        it('should go to the next state if there is only one result',
+        describe('when there is only one result', function() {
+            it('should go to the next state', function() {
+                return tester
+                    .inputs("Friend Street, South Africa")
+                    .check.interaction({
+                        state:'states:end'
+                    })
+                    .check(function(api) {
+                        var contact = api.contacts.store[0];
+                        assert.equal(contact.extra['location:formatted_address'],
+                                     'Friend Street, Cape Town 7925, South Africa');
+                    })
+                    .run();
+            });
+            it('should pass the address to the next function', function() {
+                var address_chosen;
+                tester.data.opts.next = function(address) {
+                    address_chosen = address;
+                    return 'states:end';
+                };
+                return tester
+                    .inputs("Friend Street, South Africa")
+                    .check.interaction({
+                        state:'states:end'
+                    })
+                    .check(function(api) {
+                        assert(
+                            address_chosen instanceof AddressResult,
+                            "Expected an AddressResult");
+                        assert.strictEqual(
+                            address_chosen.label,
+                            'Friend Street, Cape Town 7925, South Africa');
+                    })
+                    .run();
+            });
+        });
+
+        it('should go to the next state when location is selected from list',
         function() {
             return tester
-                .inputs("Friend Street, South Africa")
+                .inputs("Friend Street", '1')
                 .check.interaction({
                     state:'states:end'
                 })
                 .check(function(api) {
                     var contact = api.contacts.store[0];
                     assert.equal(contact.extra['location:formatted_address'],
-                        'Friend Street, Cape Town 7925, South Africa');
+                            'Friend Street, Amesbury, MA 01913, USA');
                 })
                 .run();
         });
```

Comment:
Thanks for catching that -- I think I must have somehow duplicated the test while re-organizing things.